### DISCUSSION TITLE
how to deply my orchardcore cms to linux debian?

### DISCUSSION THREAD

**Discussion by MRSheikh** [OrchardCMS/OrchardCore#8208] [Q&A] [answered]
hi.
i had created my Orchardcore Cms with 2 modules on windows 10 and want to deploy it on Linux Debian 10 . i copied publish folder and restore my database backup into my Debian server. but when i run my project on port 5000 , the setup page of orchardcore Cms appears!!!. i check i copied App_Data folder on my server too. but only  the setup page of orchardcore cms appears.
thanks for reading.

**Comment by Skrypt** [comment]
Have you moved your Data Protection folder/files too?

**Reply by MRSheikh** [reply]
Yes. I move all files and folders into app_data

**Reply by Skrypt** [reply]
It probably doesn't find your App_Data folder. 
Try to log this : `Server.MapPath("~/App_Data");`

**Reply by Skrypt** [reply]
Also you can change this path by assigning the ORCHARD_APP_DATA environment variable of your system.
See https://hub.docker.com/r/orchardproject/orchardcore-cms-linux

**Reply by MRSheikh** [reply]
> 
> 
> Also you can change this path by assigning the ORCHARD_APP_DATA environment variable of your system.
> See https://hub.docker.com/r/orchardproject/orchardcore-cms-linux

thanks Skrypt but after that this error occurred:
```
An unhandled exception occurred while processing the request.
NullReferenceException: Object reference not set to an instance of an object.

OrchardCore.ContentManagement.Display.ContentItemDisplayManager.BuildDisplayAsync(ContentItem contentItem, IUpdateModel updater, string displayType, string groupId) in ContentDisplayManager.cs, line 65
```

**Comment by MRSheikh** [comment]
thanks Skrypt but after that this error occurred:
```
An unhandled exception occurred while processing the request.
NullReferenceException: Object reference not set to an instance of an object.

OrchardCore.ContentManagement.Display.ContentItemDisplayManager.BuildDisplayAsync(ContentItem contentItem, IUpdateModel updater, string displayType, string groupId) in ContentDisplayManager.cs, line 65
```
![image](https://user-images.githubusercontent.com/17869166/103451568-b8169000-4cda-11eb-86c2-24eb0a508f65.png)

**and when i create new  site with new database and with my theme recipe this error occurred:** 
![image](https://user-images.githubusercontent.com/17869166/103452712-7d672480-4ce7-11eb-9bf5-381e1b62c2c2.png)

**Comment by Skrypt** [comment]
Make sure you have read permission on this folder. It needs to read the DataProtection files. I'm guessing here, I would need to look at the code and reproduce it. Have you tried to create a docker image on your Debian server instead. Just create a docker container and test it instead of configuring the Debian server directly. I will try to make this test if I get time. Though, one thing that is sure is that it works perfectly fine under Ubuntu which is based on Debian.

**Reply by MRSheikh** [reply]
**it's case sensitivity of table names !!!!**

**Comment by MRSheikh** [comment, marked as answer]
🥇 💯 After couple of days i found the symptom that cause this issue.  **`it's case sensitivity of table names`**
i was installed OrchardCore in windows and deploy it on Linux Debian. so in windows orchardcore table names are not case sensitive and i see all table names lowercase but when i deploy my project to Linux, orchardcore try to connect table names with PascalCase and it's not found tables. 🥇 💯 
i do this on linux to manipulate table names with case insensitivity :
```
Open terminal and edit /etc/mysql/my.cnf

sudo nano /etc/mysql/my.cnf

Underneath the [mysqld] section.add:

lower_case_table_names = 1

Restart mysql

sudo /etc/init.d/mysql restart
```
